### DISCUSSION TITLE
Can I update the version of undici which is included by hardhat?

### DISCUSSION THREAD

**Discussion by Ryan-Guo-m** [NomicFoundation/hardhat#3770] [Troubleshooting] [answered]
Hi! I had a trouble in my project recently.
```shell
ConnectTimeoutError: Connect Timeout Error
    at onConnectTimeout (/home/guo/hh-fcc/hardhat-fund-me-fcc/node_modules/hardhat/node_modules/undici/lib/core/connect.js:182:24)
    at /home/guo/hh-fcc/hardhat-fund-me-fcc/node_modules/hardhat/node_modules/undici/lib/core/connect.js:129:46
    at Immediate._onImmediate (/home/guo/hh-fcc/hardhat-fund-me-fcc/node_modules/hardhat/node_modules/undici/lib/core/connect.js:170:9)
    at processImmediate (node:internal/timers:466:21) {
  code: 'UND_ERR_CONNECT_TIMEOUT'
}
```
Pay attention to this: `/hardhat-fund-me-fcc/node_modules/hardhat/node_modules/undici`.
I tried many different methods, for example, switching networks, running projects at different points in time, and so on, but nothing works. 
After that,  I have a sudden inspiration. It could be a problem with undici, but it's included by hardhat.
So, can I change the version of undici plugin which is included by hardhat plugin？ And if I could do that, how should I do?
Thank you for your answer!

**Comment by fvictorio** [comment, marked as answer]
You should be able to do it using [resolutions](https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/) (if you are using yarn v1) or [overrides](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#overrides) (if you are using a version of npm that supports them; I'm not sure which version added them).

**Reply by Ryan-Guo-m** [reply]
Ok. Thank you so much!
